File: 200_LeetCode_Questions/C++/D12_P1_Leetcode_977.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& A) {
        int i,j;
        int n= A.size();
        for(i=0;i<n-1;i++){
            if(A[i+1]>0)
                break;
        }
        j=i+1;
        vector<int> ans(n);
        int k=0;
        while(i>=0 && j<n){
            if(abs(A[i])<A[j]){
                ans[k++]=pow(A[i--],2);
            }
            else
                ans[k++]=pow(A[j++],2);
        }
        while(i!=-1){
            ans[k++]=pow(A[i--],2);
        }
        while(j!=n)
            ans[k++]=pow(A[j++],2);
        return ans;
    }
};

File: 200_LeetCode_Questions/Java/D12P1_LeetCode_977.java
/*
Runtime: 1 ms, faster than 100.00% of Java online submissions for Squares of a Sorted Array.
Memory Usage: 40.8 MB, less than 83.69% of Java online submissions for Squares of a Sorted Array.
*/

class Solution {
    public int[] sortedSquares(int[] A) {
        int[] returnedArray = new int[A.length];
        int right = A.length-1;
        int left = 0;
        for(int i=0 ; i < A.length ; i++){
            if(Math.abs(A[left])<Math.abs(A[right])){
                returnedArray[A.length-i-1] = A[right]*A[right];
                right--;
            }else{
                returnedArray[A.length-i-1] = A[left]*A[left];
                left++;
            }
        }
        return returnedArray;
    }
}